### ISSUE TITLE
Build pdfjs into a single viewer.html page (all images, js, etc embedded)? 

### ISSUE THREAD

**Issue by connwork** [mozilla/pdf.js#16865]
Hi, 

I've seen references to single-file builds scattered across the internet, but no real insight into how to achieve this. 

Is there any way to build the default viewer into a single file? Trying this manually leads to several issues I have encountered, including issues with annotation images, etc. 

Related issue here that was immediately closed and the author was not responded to: https://github.com/mozilla/pdf.js/issues/14781

**Comment by Snuffleupagus** [comment]
Sorry, but this is not something that we can provide support/help with unfortunately. Please keep in mind that the primary development target for this library is the *built-in* Firefox PDF Viewer.

**Comment by connwork** [comment]
It would be easier and would not require support if there was actually documentation.

**Comment by Snuffleupagus** [comment]
> It would be nice if some users who managed to solve these kind of issues could make a PR to improve the documentation.

Honestly, I don't think that the particular use-case in this issue is one what we can/should support since it's too niche and considering that all code needs to be maintained (the latter point very often becomes a problem for edge-case features).
Furthermore note that we *purposely* removed the "single file" build-target, which bundled the `pdf.js` and `pdf.worker.js` files, years ago because it performed poorly; hence bringing something like that back seems generally undesirable.

**Comment by connwork** [comment]
I'm developing a custom solution based on PDF.js core library that requires a single directory of files with as few files as possible, all renamed with a specific naming scheme so that PDF.js can interact with a second program. 

I've sunk hours into attempting to get this to work, and I do have a barely functioning prototype that could be an incredible open source resource. 

I've attempted multiple times at paying people to help me, but all of them take one look at this repo and tell me it's not worth it because of the lack of documentation and how long it would take them to familiarize themselves with the code. Trying to find help from someone that already is familiar with pdfjs also seems impossible.

I'm not continuing this conversation because I have any hope that you will now help me after I have presented these uncomfortable truths. My only goal is that hopefully someone from your team will realize how inhibitory this is for developers and take steps to change it.
